src/userinput.c

```diff
@@ -135,6 +135,7 @@ static int pinentry_read(int from, char **retstr)
 	    || strncmp(buf, "D ", 2) == 0) {
 		if (retstr) {
 			*retstr = strchr(buf, ' ');
+			*retstr = strtok(*retstr, "\n");
```

Comment:
Use **`strtok_r`** instead of `strtok`. See #571.

Reply:
```suggestion
			*retstr = strtok_r(*retstr, "\n");
```

Reply:
O, I was not aware

Reply:
[Function `strtok_r` requires 3 arguments](https://man7.org/linux/man-pages/man3/strtok_r.3.html).

Reply:
OK, sorry. I didn't test.

```suggestion
                        char *saveptr = NULL;
			*retstr = *retstr ? strtok_r(*retstr, "\n", &saveptr) : NULL;
```

Reply:
I'll update the fork and make a new merge request.